profiles/permissions.py

```diff
@@ -15,3 +16,15 @@ class ReadOnly(BasePermission):
     def has_permission(self, request, view):
         if request.method in SAFE_METHODS:
                  return True
+
+
+class SelfCompany(BasePermission):
+    def has_permission(self, request, view):
+        user = request.user
+        pk = request.data.get("company_pk")
+        try:
+            profile = Profile.objects.get(id=pk)
+        except:
```

Comment:
Never EVER use `try  .. except` without limiting it to particular exception. `ObjectDoesNotExist` should be suitable here.

Reply:
Thank you for feedback. Done.